source/adapters/level_zero/command_buffer.cpp

```diff
@@ -1087,6 +1227,21 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferEnqueueExp(
   ZE2UR_CALL(zeCommandQueueExecuteCommandLists,
              (ZeCommandQueue, 1, &CommandBuffer->ZeCommandList, ZeFence));
 
+  // The Copy command-list is submitted to the main copy queue iif it is not
+  // empty.
+  if (!CommandBuffer->MCopyCommandListEmpty) {
+    ze_fence_handle_t ZeCopyFence;
+    ZeStruct<ze_fence_desc_t> ZeCopyFenceDesc;
+    auto &QGroupCopy = Queue->getQueueGroup(true);
+    uint32_t QueueGroupOrdinal;
+    auto &ZeCopyCommandQueue = QGroupCopy.getZeQueue(&QueueGroupOrdinal);
+    ZE2UR_CALL(zeFenceCreate, (ZeCopyCommandQueue, &ZeFenceDesc, &ZeCopyFence));
```

Comment:
Why do we need to return a Fence here? I don't think we are synchronizing that fence anywhere right? 

I have similar comment about line 1227 but that's already existing code. I removed that fence in this PR with no immediately visible drawbacks: https://github.com/oneapi-src/unified-runtime/pull/1709/files

Reply:
I had forgotten about that PR. I think that makes Fence for the compute command queue useful. 

Still unsure about the fence for the copy command queue. Sounds like it's dead code until we support updating memory operation nodes?

Reply:
Good point, until we can update commands in a copy command-list I don't think we need to use the fence for this command-list in the same way inside our update implementation. So think we could omit creating and using an event here.